src/ripple/app/main/Application.cpp

```diff
@@ -1976,6 +1976,17 @@ bool ApplicationImp::loadOldLedger (
                 }
             }
         }
+        using namespace std::chrono_literals;
+        static constexpr char ledgerWarnTime[] {"Jan 01, 2018 00:00:00 UTC"};
+        static constexpr NetClock::time_point ledgerWarnTimePoint {568080000s};
+        if (loadLedger->info().closeTime < ledgerWarnTimePoint)
+        {
+            JLOG(m_journal.fatal()) <<
+                "\n\n*** You are replaying a ledger from before " <<
+                ledgerWarnTime << " ***\n"
+                "*** You may want to run an earlier version of rippled ***\n"
+                "*** that supports the older rules.  Continuing. ***\n";
+        }
```

Comment:
@seelabs, how would you feel about just refusing to process old ledgers?  Personally, I can go either way since I seldom replay old ledgers.  But I can imagine situations where we want to replay an old ledger with newer code to see whether an old bug is fixed.  In those cases blocking old ledger replay would be counterproductive.

But I'll implement whatever you two agree to.

Reply:
I'm OK with allowing such replay **if** we make it clear that the result may be different because of different rules. I don't think the message in place now achieves that.

Reply:
@scottschurr It's not a killer. Replaying old transactions is rare enough that I can hack the code myself. I understand Nik's concern about not replaying ledgers with different rules. I'm OK refusing to replay.

Reply:
@seelabs, I'm going with a new-and-improved warning that @nbougalis and I have agreed to:
```
***  WARNING   ***
You are replaying a ledger from before 2018-Jan-01 00:00:00.000000 UTC.
This replay will not handle your ledger as it was originally handled.
Consider running an earlier version of rippled to get the older rules.
*** CONTINUING ***
```